tests/integration/springer/test_dag_process_file.py

```diff
@@ -47,39 +52,56 @@ def extract_zip_to_article(zip_filename):
     return article
 
 
-def test_dag_loaded(dag: DAG):
-    assert dag is not None
-    assert len(dag.tasks) == 4
-
-
-def test_dag_run(dag: DAG, article):
-    id = datetime.datetime.utcnow().strftime(
-        "test_springer_dag_process_file_%Y-%m-%dT%H:%M:%S.%f"
-    )
-    dagrun = dag.create_dagrun(
-        DagRunState.QUEUED,
-        run_id=id,
-        conf={"file": base64.b64encode(ET.tostring(article)).decode()},
-    )
-    wait().at_most(60, SECOND).until(lambda: __test_dagrun_state(dagrun))
-
-
-def test_dag_run_no_input_file(dag: DAG):
-    id = datetime.datetime.utcnow().strftime(
-        "test_springer_dag_process_file_%Y-%m-%dT%H:%M:%S.%f"
-    )
-    dagrun = dag.create_dagrun(DagRunState.QUEUED, run_id=id)
-    wait().at_most(60, SECOND).until(lambda: __test_dagrun_fails(dagrun))
-
-
-def test_dag_parse_file(article):
-    springer_parse_file(
-        params={"file": base64.b64encode(ET.tostring(article)).decode()}
-    )
+class TestClassSpringerFilesHarvesting:
+    def test_dag_loaded(self, dag: DAG):
+        assert dag is not None
+        assert len(dag.tasks) == 4
+
+    def test_dag_run(self, dag: DAG, dag_was_paused: bool, article: ET):
+        id = datetime.datetime.utcnow().strftime(
+            "test_springer_dag_process_file_%Y-%m-%dT%H:%M:%S.%f"
+        )
+        if dag.get_is_paused():
+            DagModel.get_dagmodel(dag.dag_id).set_is_paused(is_paused=False)
+        dagrun = dag.create_dagrun(
+            DagRunState.QUEUED,
+            run_id=id,
+            conf={"file": base64.b64encode(ET.tostring(article)).decode()},
+        )
+        wait().at_most(60, SECOND).until(lambda: self.__test_dagrun_state(dagrun))
+        if dag_was_paused:
+            DagModel.get_dagmodel(dag.dag_id).set_is_paused(is_paused=True)
+
+    def test_dag_run_no_input_file(self, dag: DAG, dag_was_paused: bool):
+        if dag.get_is_paused():
+            DagModel.get_dagmodel(dag.dag_id).set_is_paused(is_paused=False)
+        id = datetime.datetime.utcnow().strftime(
+            "test_springer_dag_process_file_%Y-%m-%dT%H:%M:%S.%f"
+        )
+        dagrun = dag.create_dagrun(DagRunState.QUEUED, run_id=id)
+        wait().at_most(60, SECOND).until(lambda: self.__test_dagrun_fails(dagrun))
+        if dag_was_paused:
+            DagModel.get_dagmodel(dag.dag_id).set_is_paused(is_paused=True)
+
+    def test_dag_parse_file(self, article):
+        springer_parse_file(
+            params={"file": base64.b64encode(ET.tostring(article)).decode()}
+        )
+
+    def __test_dagrun_fails(self, dagrun):
+        dagrun.update_state()
+        return not dagrun.get_state() == DagRunState.FAILED
+
+    def __test_dagrun_state(self, dagrun):
```

Comment:
As I wrote, I would move it to another module. Also, it's a bit confusing for me: can a dag have more than 1 state? So if it's running for example it's not possible that it's failed or queued? If so, maybe you can have one assertion function where you can pass the desired state and assert it?

Reply:
No, the DAG **run** cannot have more than one state at a time